searching from pubmed or arxiv doesn't work

**Describe the bug**
searching from external providers like pubmed or arxiv doesn't work. The issue seems to be related to a dependencies issue with `jersey-core` library, and is similar to what reported in [this comment of the remove BTE PR](https://github.com/DSpace/DSpace/pull/3189#pullrequestreview-628720183). Here the error stacktrace 

```
Caused by: java.lang.NoSuchMethodError: 'java.lang.Object javax.ws.rs.core.Response.readEntity(java.lang.Class)'
	at org.dspace.importer.external.pubmed.service.PubmedImportMetadataSourceServiceImpl$GetRecords.call(PubmedImportMetadataSourceServiceImpl.java:302) ~[dspace-api-7.0-beta5-SNAPSHOT.jar:7.0-beta5-SNAPSHOT]
	at org.dspace.importer.external.pubmed.service.PubmedImportMetadataSourceServiceImpl$GetRecords.call(PubmedImportMetadataSourceServiceImpl.java:262) ~[dspace-api-7.0-beta5-SNAPSHOT.jar:7.0-beta5-SNAPSHOT]
	at org.dspace.importer.external.service.components.AbstractRemoteMetadataSource.retry(AbstractRemoteMetadataSource.java:171) ~[dspace-api-7.0-beta5-SNAPSHOT.jar:7.0-beta5-SNAPSHOT]
	at org.dspace.importer.external.pubmed.service.PubmedImportMetadataSourceServiceImpl.getRecords(PubmedImportMetadataSourceServiceImpl.java:108) ~[dspace-api-7.0-beta5-SNAPSHOT.jar:7.0-beta5-SNAPSHOT]
	at org.dspace.external.provider.impl.LiveImportDataProvider.searchExternalDataObjects(LiveImportDataProvider.java:97) ~[dspace-api-7.0-beta5-SNAPSHOT.jar:7.0-beta5-SNAPSHOT]
	at org.dspace.external.service.impl.ExternalDataServiceImpl.searchExternalDataObjects(ExternalDataServiceImpl.java:65) ~[dspace-api-7.0-beta5-SNAPSHOT.jar:7.0-beta5-SNAPSHOT]
	at org.dspace.app.rest.repository.ExternalSourceRestRepository.getExternalSourceEntries(ExternalSourceRestRepository.java:73) ~[classes/:7.0-beta5-SNAPSHOT]
	at org.dspace.app.rest.repository.ExternalSourceRestRepository$$FastClassBySpringCGLIB$$62b12abd.invoke(<generated>) ~[classes/:7.0-beta5-SNAPSHOT]
	at org.springframework.cglib.proxy.MethodProxy.invoke(MethodProxy.java:218) ~[spring-core-5.2.5.RELEASE.jar:5.2.5.RELEASE]
	at org.springframework.aop.framework.CglibAopProxy$DynamicAdvisedInterceptor.intercept(CglibAopProxy.java:687) ~[spring-aop-5.2.5.RELEASE.jar:5.2.5.RELEASE]
	at org.dspace.app.rest.repository.ExternalSourceRestRepository$$EnhancerBySpringCGLIB$$94170eb9.getExternalSourceEntries(<generated>) ~[classes/:7.0-beta5-SNAPSHOT]
	at org.dspace.app.rest.ExternalSourcesRestController.getExternalSourceEntries(ExternalSourcesRestController.java:64) ~[classes/:7.0-beta5-SNAPSHOT]
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:?]
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:?]
	at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:?]
	at java.lang.reflect.Method.invoke(Method.java:566) ~[?:?]
	at org.springframework.web.method.support.InvocableHandlerMethod.doInvoke(InvocableHandlerMethod.java:190) ~[spring-web-5.2.5.RELEASE.jar:5.2.5.RELEASE]
	at org.springframework.web.method.support.InvocableHandlerMethod.invokeForRequest(InvocableHandlerMethod.java:138) ~[spring-web-5.2.5.RELEASE.jar:5.2.5.RELEASE]
	at org.springframework.web.servlet.mvc.method.annotation.ServletInvocableHandlerMethod.invokeAndHandle(ServletInvocableHandlerMethod.java:105) ~[spring-webmvc-5.2.5.RELEASE.jar:5.2.5.RELEASE]
	at org.springframework.web.servlet.mvc.method.annotation.RequestMappingHandlerAdapter.invokeHandlerMethod(RequestMappingHandlerAdapter.java:879) ~[spring-webmvc-5.2.5.RELEASE.jar:5.2.5.RELEASE]
	at org.springframework.web.servlet.mvc.method.annotation.RequestMappingHandlerAdapter.handleInternal(RequestMappingHandlerAdapter.java:793) ~[spring-webmvc-5.2.5.RELEASE.jar:5.2.5.RELEASE]
	at org.springframework.web.servlet.mvc.method.AbstractHandlerMethodAdapter.handle(AbstractHandlerMethodAdapter.java:87) ~[spring-webmvc-5.2.5.RELEASE.jar:5.2.5.RELEASE]
	at org.springframework.web.servlet.DispatcherServlet.doDispatch(DispatcherServlet.java:1040) [spring-webmvc-5.2.5.RELEASE.jar:5.2.5.RELEASE]
	... 103 more
```

**To Reproduce**
Go to the following hal browser links that provide requests that should provide a result from an external source provider :  
- pubmed example https://api7.dspace.org/server/#/server/api/integration/externalsources/pubmed/entries?sort=score,DESC&page=0&size=10&query=30031392
- arxiv example https://api7.dspace.org/server/#https://api7.dspace.org/server/api/integration/externalsources/arxiv/entries?sort=score,DESC&page=0&size=10&query=test


**Expected behavior**
Both requets return an 500 status code, but should return the result list or a not found status

--- a/dspace-api/pom.xml
+++ b/dspace-api/pom.xml
@@ -856,6 +856,10 @@
                     <groupId>org.javassist</groupId>
                     <artifactId>javassist</artifactId>
                 </exclusion>
+                <exclusion>
+                	<groupId>io.swagger</groupId>
+                	<artifactId>swagger-jersey-jaxrs</artifactId>
+                </exclusion>
             </exclusions>
         </dependency>
 

